framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,286 +85,180 @@ private static Pattern anchored(String regex) {
+        } catch (ParseProblemException e) {
+            throw constructParserException(expression, "is an invalid expression");
+        }
+        Receiver result = getReceiverFromExpression(expr, context, localScope);
         if (result instanceof ClassName && !expression.endsWith("class")) {
             throw constructParserException(
                     expression, "a class name cannot terminate a flow expression string");
         }
         return result;
     }
 
-    private static FlowExpressions.Receiver parseHelper(
-            String expression, FlowExpressionContext context, TreePath path)
+    /**
+     * Converts the passed Javaparser Expression to a {@link Receiver}.
+     *
+     * @return the Receiver of the passed expression
+     */
+    private static Receiver getReceiverFromExpression(
+            Expression expr, FlowExpressionContext context, TreePath path)
             throws FlowExpressionParseException {
-        expression = expression.trim();
 
         ProcessingEnvironment env = context.checkerContext.getProcessingEnvironment();
         Types types = env.getTypeUtils();
 
-        if (isNullLiteral(expression, context)) {
-            return parseNullLiteral(types);
-        } else if (isIntLiteral(expression, context)) {
-            return parseIntLiteral(expression, types);
-        } else if (isLongLiteral(expression, context)) {
-            return parseLongLiteral(expression, types);
-        } else if (isFloatLiteral(expression, context)) {
-            throw constructParserException(
-                    expression,
-                    String.format("floating-point values '%s' cannot be parsed", expression));
-        } else if (isStringLiteral(expression, context)) {
-            return parseStringLiteral(expression, types, env.getElementUtils());
-        } else if (isThisLiteral(expression, context)) {
-            return parseThis(context);
-        } else if (isSuperLiteral(expression, context)) {
-            return parseSuper(expression, types, context);
-        } else if (isIdentifier(expression)) {
-            return parseIdentifier(expression, env, path, context);
-        } else if (isParameter(expression, context)) {
-            return parseParameter(expression, context);
-        } else if (isArray(expression)) {
-            return parseArray(expression, context, path);
-        } else if (isMethodCall(expression)) {
-            return parseMethodCall(expression, context, path, env);
-        } else if (isMemberSelect(expression)) {
-            return parseMemberSelect(expression, env, context, path);
-        } else if (isParentheses(expression, context)) {
-            return parseParentheses(expression, context, path);
-        } else {
-            String message;
-            if (expression.equals("#0")) {
-                message =
-                        "one should use \"this\" for the receiver or \"#1\" for the first formal parameter";
+        if (expr != null) {
```

Comment:
The call to this method issue an error if `expr` is null.  That way this method never returns null.